### DISCUSSION TITLE
Prisma Database conection queue full

### DISCUSSION THREAD

**Discussion by lukadriel7** [prisma/prisma#6660] [Q&A] [answered]
I am currently trying to import a large dataset into my database using prisma.
I get my data from a csv file which has  more than 15,000 rows.
unfortunately I run into the following error:

```
PrismaClientKnownRequestError2 [PrismaClientKnownRequestError]: 
Invalid `prisma.medicament.create()` invocation:


  Failed to run future: no available capacity
    at cb (/home/user/Documents/VSCode/Altea/DPI/api/node_modules/@prisma/client/runtime/index.js:34780:17)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async Parser.<anonymous> (/home/user/Documents/VSCode/Altea/DPI/api/dist/app.service.js:97:32) {
  code: 'QueueFull',
  clientVersion: '2.21.2',
  meta: undefined

```
This is the code I am using:

```

import { createReadStream } from "fs";
import { parse as csvParse } from "csv";
import { parse as dateParse } from "date-fns";

const stream = createReadStream(
            "/home/user/Documents/VSCode/CIS_MEDS_UTF-8/CIS_bdpm/CIS_bdpm_1.txt",
            {
                encoding: "utf8",
            }
        );

        stream
            .on("error", (err) => {
                console.log("error stream: ", err);
            })
            .pipe(
                csvParse({
                    delimiter: "\t",
                    //ignore_last_delimiters: false,
                    //record_delimiter: "\t\r\n",
                    cast: (value, context) => {
                        if (context.index === 0) {
                            return parseInt(value);
                        } else if (context.index === 7) {
                            return dateParse(value, "dd/MM/yyyy", new Date());
                        } else if (context.index === 11) {
                            return value === "oui" ? true : false;
                        } else return value;
                    },
                    columns: [
                        "code_cis",
                        "denomination",
                        "forme_pharmaceutique",
                        "voie_administration",
                        "statut_administratif",
                        "type_procedure_autorisation",
                        "etat_commercialisation",
                        "date_amm",
                        "statut_bdm",
                        "numero_autorisation",
                        "titulaire",
                        "surveillance_renforcee",
                    ],
                })
            )
            .on("data", async (row) => {
                try {
                    const result = await this.prismaService.medicament.create({
                        data: row,
                    });
                } catch (error) {
                    console.error(`while inserting ${row.code_cis} in db: `, error);
                }
            });

```

From my understanding, the queue is full, is there anyway to increase the queue size of maybe a recommendation  to optimize my code ?
Thanks in advance

**Comment by janpio** [comment, marked as answer]
I think you might have exhausted an internal limit in our binary - this queue is not something you should normally be informed about.

Can you please open a bug issue and share your schema file and possible also the data file you are importing? Then we could try to reproduce this.

Now to your problem: Could you possibly use [`createMany`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#createmany)?

**Reply by lukadriel7** [reply]
Hello, sorry for the late reply since I am loading the file line by line, I didn't think of using createMany,
I will have to load a good portion of the file to do so. I will open an issue and share my schema and the file i am using.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
